### PR TITLE
Fixes and improvements to AKS cluster creation script

### PR DESCRIPTION
**Pull request by yuchaoran2011**
### What changes were proposed in this pull request?
* The latest fully supported version of Kubernetes on AKS is 1.16.9. Higher versions are also available but are still in preview. Updated the default version to be 1.16.9
* A8 VMs will be deprecated next year. Changed the default VM type to A4.
* Enable autoscaler when creating the cluster.
* Fixed a bug where the values of the parameters could be not set.

### Why are the changes needed?
Bug fixes.

### Does this PR introduce any user-facing change?
Yes. Changes are in a user-facing script.

### How was this patch tested?
Running the script.

### CODE REVIEW

File: installer/create-cluster-aks.sh [hunk 1]
```diff
@@ -45,30 +45,30 @@ function parse_args {
...
+      -r|--resource-group)
         shift
         RESOURCE_GROUP_NAME=$1
         [ -z "$RESOURCE_GROUP_NAME" ] && error '"--resource-group" requires a non-empty option argument.\n'
       ;;
       --location)
         shift
         LOCATION=$1
-        [ -z "$LOCATION" ] && error LOCATION=eastus
+        [ -z "$LOCATION" ] && error '"--location" requires a non-empty option argument.\n'
```
Comment (maasg) [inline installer/create-cluster-aks.sh:56]: I don't completely understand the phrase "a non-empty option argument."
Do you mean "a non-empty argument." ? 
(same x3 below)

File: installer/create-cluster-aks.sh [hunk 2]
```diff
@@ -113,9 +118,12 @@ az aks create --resource-group "$RESOURCE_GROUP_NAME" \
   --name "$CLUSTER_NAME" \
   --node-count $NODE_COUNT \
   --enable-addons monitoring \
+  --enable-cluster-autoscaler \
   --generate-ssh-keys \
   --kubernetes-version "$CLUSTER_VERSION" \
-  --node-vm-size $NODE_VM_SIZE
+  --node-vm-size $NODE_VM_SIZE \
+  --min-count 1 \
+  --max-count 7
```
Comment (maasg) [inline installer/create-cluster-aks.sh:126]: How are min/max count related to `$NODE_COUNT` ? 
e.g. if the user gives a $NODE_COUNT > max-count, what's the expected behavior?

File: installer/create-cluster-aks.sh [hunk 3]
```diff
@@ -45,30 +45,30 @@ function parse_args {
         CLUSTER_NAME=$1
         [ -z "$CLUSTER_NAME" ] && error '"--cluster-name" requires a non-empty option argument.\n'
```
Comment (RayRoestenburg) [inline installer/create-cluster-aks.sh:46]: I think it is better to remove the "option" in these sentences. Can't we just say `--clustername requires an argument?`?

File: installer/create-cluster-aks.sh (hunk 1 above)
Comment (RayRoestenburg) [reply installer/create-cluster-aks.sh:56]: I think I would prefer ```--location" requires an argument.\n```

File: installer/create-cluster-aks.sh (hunk 2 above)
Comment (yuchaoran2011) [reply installer/create-cluster-aks.sh:126]: Good question. I haven't tested that scenario yet. Our GKE cluster creation script has the same issue. I'll update both scripts to check that node count is between and min and max counts.

File: installer/create-cluster-aks.sh (hunk 3 above)
Comment (yuchaoran2011) [reply installer/create-cluster-aks.sh:46]: Agreed. Will change.

File: installer/create-cluster-aks.sh (hunk 2 above)
Comment (yuchaoran2011) [reply installer/create-cluster-aks.sh:119]: In the script for GKE, we hard code the min, max and default node counts to be 1, 7 and 5, respectively. I updated this script to do the same. The user can go in the script and change the numbers if they want.

**Pull request merged**